test/integration/scheduler_perf/scheduler_perf_test.go

```diff
@@ -71,11 +71,17 @@ const (
 
 var (
 	defaultMetricsCollectorConfig = metricsCollectorConfig{
-		Metrics: []string{
-			"scheduler_scheduling_algorithm_predicate_evaluation_seconds",
-			"scheduler_scheduling_algorithm_priority_evaluation_seconds",
-			"scheduler_e2e_scheduling_duration_seconds",
-			"scheduler_pod_scheduling_duration_seconds",
+		// Specify the filter as {"label": "value"} to filter out the specific metrics
+		// Specify "*" as {"label": "*"} denotes that all the metrics with the label should be collected.
+
+		// AggregatedMetric defines the metrics that will aggregate all the Histogram data within the same metricFamily.
+		AggregatedMetric: map[string]map[string]string{
+			"scheduler_e2e_scheduling_duration_seconds": nil,
+			"scheduler_pod_scheduling_duration_seconds": nil,
+		},
+		// HistogramMetric defines the metrics that will collect each of Histogram from the metricFamily one by one.
+		HistogramMetric: map[string]map[string]string{
+			"scheduler_framework_extension_point_duration_seconds": {"extension_point": "*"},
```

Comment:
Instead of adding another parsing layer here, I'd expect to just use prometheus-compatible strings here. And then use standard promethues library to parse it.

- `scheduler_framework_extension_point_duration_seconds` - the parsed result should be a histogram vector
- `scheduler_framework_extension_point_duration_seconds[extension_point="Filter"]` - the parsed result should be a single histogram

The good thing is we don't really care about whether the input is a vector or not, the parsed result should be able to tell.

And once we have the parsed result, we can compose  our custom HistogramVector easily:

- if it's a single histogram, just compose our custom HistogramVector with one element
- if it's a histogram vector, just unbox/box into our custom HistogramVector

With this, we also don't need another set of functions of Histogram - the existing functions HistogramVector should work for both scenarios.

Do you think it's feasible?

Reply:
> Instead of adding another parsing layer here

The layer is added to honor the `lvMap` parameter, or else I cannot see how `lvMap` could be used.

> And once we have the parsed result, we can compose our custom HistogramVector easily:

> if it's a single histogram, just compose our custom HistogramVector with one element
> if it's a histogram vector, just unbox/box into our custom HistogramVector

The question is we cannot unbox the vector which is collected from `GetHistogramVecFromGatherer`, the struct is not applicable for the case where we want to collect each of the Histogram from the vector.

Reply:
IIUC, the introduced `HistogramMetric` is to display a HistogramVec's items **individually**, and also **dynamically**. Say, if I specify `extension_point: *`, it will show a list of Histograms on PreFilter, Filter, Score, etc.

If that's the case, it doesn't look like exactly our original requirement. The original requirement was to show the Histrogram of Filter and Score only. That should be able to handled by:

- "scheduler_framework_extension_point_duration_seconds": {"extension_point": "Filter"}
- "scheduler_framework_extension_point_duration_seconds": {"extension_point": "Score"}

with the type `AggregatedMetric`, right?

Reply:
My point is that some of the changes introduced in this PR is to implement a "new" requirement. We should get to a consensus if we need that feature first. If we don't need that, we're making the logic unnecessarily complex.

cc @alculquicondor

Reply:
true, the initial purpose was as what this issue stated: https://github.com/kubernetes/kubernetes/issues/96252, before  the removing of "scheduler_scheduling_algorithm_predicate_evaluation_seconds" and "scheduler_scheduling_algorithm_priority_evaluation_seconds" I remembered we were collecting each of them individually, and all those metrics are seems necessary to be provided by the tool. So back at that time, I pushed the PR in order to fix the gap.

but with the PR already merged by @Huang-Wei , this could be still addressed define multiple items,
"scheduler_framework_extension_point_duration_seconds": {"extension_point": "Filter"}
"scheduler_framework_extension_point_duration_seconds": {"extension_point": "Sore"}

the drawback is obviously, 
- it's not effective enough (n times to call the GetHistogramVecFromGatherer to fetch the data) if we want to collect all those metrics for "extension_point".
- we will not collect all those metrics for each of "extension_point" by default.

but get rid of the extra function also reduce the complexity, which make the impl more neat.

Reply:
I'm fine with the new requirement. I think it will be helpful to have all the information per extension point.

Now, I don't think we need all the flexibility that this PR is trying to implement, but I'm fine with it.